packages/mobile/src/qrcode/utils.ts

```diff
@@ -141,9 +142,17 @@ export function* handleBarcode(
       }
   yield put(storeLatestInRecents(recipient))
 
-  if (secureSendTxData) {
-    replace(Screens.SendConfirmation, { transactionData: secureSendTxData })
+  if (secureSendTxData && isOutgoingPaymentRequest) {
+    replace(Screens.PaymentRequestConfirmation, {
+      transactionData: secureSendTxData,
+      addressJustValidated: true,
+    })
+  } else if (secureSendTxData) {
+    replace(Screens.SendConfirmation, {
+      transactionData: secureSendTxData,
+      addressJustValidated: true,
+    })
   } else {
-    replace(Screens.SendAmount, { recipient })
+    replace(Screens.SendAmount, { recipient, isFromScan: true })
```

Comment:
Just wondering if the other two `replace` calls need this isFromScan prop too?

Reply:
Not needed. `isFromScan` is purely to indicate whether or not the transaction was initiated via QR scan for analytics logging. The secure send flow can never be initiated via scan (not to be confused with validating an address via scan) because QR codes are assumed to always be "correct"